forge-kernel/include/forge/Topology.hpp

```
@@ -0,0 +1,37 @@
+struct TopoSignature {
+    long vertexCount = 0;   // welded, not raw
+    long edgeCount   = 0;   // unique welded edges
+    long faceCount   = 0;   // triangles
+    long eulerChar   = 0;   // V - E + F
+    long genus       = 0;   // max(0, round((2 - chi) / 2))
+    long shellCount  = 0;   // connected components of the welded mesh
+};
+
+// Tessellate `body` and compute its weld-betti topology signature.
+// Returns false when the body cannot be meshed (degenerate / empty).
+bool topologySignature(ShapeHandle body, TopoSignature& out,
```

Clang (14.0.6)

[warning] use a trailing return type for this function

(modernize-use-trailing-return-type)